typescript/sdk/src/core/EvmCoreModule.ts

```diff
@@ -0,0 +1,251 @@
+        multiProvider,
+        config: {
+          mailbox: mailbox.address,
+          owner: await multiProvider.getSigner(chain).getAddress(),
+        },
+      })
+    ).serialize();
+
+    // Deploy Validator announce
+    const validatorAnnounce =
+      await hyperlaneCoreDeployer.deployValidatorAnnounce(
+        chainName,
+        mailbox.address,
+      );
+
+    // Deploy Test Receipient
+    const testRecipient = await hyperlaneCoreDeployer.deployTestRecipient(
+      chainName,
+      await mailbox.defaultIsm(),
+    );
+
+    return new EvmCoreModule(multiProvider, {
+      addresses: {
+        proxyAdmin,
+        mailbox,
+        validatorAnnounce,
+        interchainAccountIsm,
+        interchainAccountRouter,
+        testRecipient,
+        ismFactoryFactories: ismFactoriesFactory[chainName],
+      },
+      chain,
+      config,
+    });
+  }
+
+  static async deployProxyAdmin(
+    chainName: string,
+    hyperlaneCoreDeployer: HyperlaneCoreDeployer,
+  ) {
+    return hyperlaneCoreDeployer.deployContract(chainName, 'proxyAdmin', []);
+  }
+
+  static async deployIsmFactories(
+    chainName: string,
+    config: DerivedCoreConfig,
+    multiProvider: MultiProvider,
+  ): Promise<{
+    ismFactory: HyperlaneIsmFactory;
+    ismFactoriesFactory: ChainMap<HyperlaneContracts<ProxyFactoryFactories>>;
+  }> {
+    const proxyFactoryDeployer = new HyperlaneProxyFactoryDeployer(
+      multiProvider,
+    );
+    const ismFactoriesFactory = await proxyFactoryDeployer.deploy({
+      [chainName]: config,
+    });
+    const ismFactory = new HyperlaneIsmFactory(
+      ismFactoriesFactory,
+      multiProvider,
+    );
+
+    return { ismFactory, ismFactoriesFactory };
+  }
+
+  /**
+   * Deploys a Mailbox and it's default ISM, hook, and required hook contracts with a given configuration.
+   *
+   * @param chain - The chain name or ID to deploy the Mailbox on.
+   * @param config - The derived core configuration for the deployment.
+   * @param proxyAdmin - The address of the proxy admin for the Mailbox contract.
+   * @param deployer - The Hyperlane core deployer instance to use for the deployment.
+   * @param factories - The Hyperlane contract factories to use for the deployment.
+   * @param multiProvider - The multi-provider instance to use for the deployment.
+   * @returns The deployed Mailbox contract instance.
+   */
+  static async deployMailbox(
+    chain: ChainNameOrId,
+    config: DerivedCoreConfig,
+    proxyAdmin: Address,
+    deployer: HyperlaneCoreDeployer,
+    factories: HyperlaneContracts<ProxyFactoryFactories>,
+    multiProvider: MultiProvider,
+  ): Promise<Mailbox> {
+    const chainName = multiProvider.getChainName(chain);
+    const domain = deployer.multiProvider.getDomainId(chainName);
+    const mailbox = await deployer.deployProxiedContract(
+      chainName,
+      'mailbox',
+      'mailbox',
+      proxyAdmin,
+      [domain],
+    );
+
+    if (typeof config.defaultIsm !== 'string') {
```

Comment:
Just a passing thought, we had a lot of these `typeof string` checks in the old deployer code and it feels brittle. It would be great if configs could **always** be objects with an explicit `type` field for discriminated unions. And if the only other value in the config is an `address: string` field, that's totally fine! cc @yorhodes @paulbalaji

Reply:
Not something you have to solve in this PR but if we're in agreement I think it should be done as part of the migration to CRUD

Reply:
sounds like a good idea to me - might be something I can look into as so far the only time I've seen the string/address check is for ISMs

Reply:
agree though that the union type everywhere is annoying

Reply:
super agree!